app/Libraries/OrderCheckout.php

```diff
@@ -38,10 +39,18 @@ class OrderCheckout
      */
     private $provider;
 
-    public function __construct(Order $order, string $provider = null)
+    /** @var string|null */
+    private $shopifyId;
+
+    public function __construct(Order $order, ?string $provider = null, ?string $shopifyId = null)
     {
+        if ($provider === Order::PROVIDER_SHOPIFY && $shopifyId === null) {
+            throw new InvariantException('shopify provider requires a checkout id.');
+        }
+
         $this->order = $order;
         $this->provider = $provider;
```

Comment:
should check it's valid provider listed in `allowedCheckoutTypes` (er, `allowedCheckoutProviders`?).